File: main/src/main/java/com/sedmelluq/discord/lavaplayer/source/youtube/YoutubeSignatureResolver.java
package com.sedmelluq.discord.lavaplayer.source.youtube;

import com.sedmelluq.discord.lavaplayer.tools.io.HttpInterface;
import java.net.URI;

public interface YoutubeSignatureResolver {
  URI resolveFormatUrl(HttpInterface httpInterface, String playerScript, YoutubeTrackFormat format) throws Exception;

  String resolveDashUrl(HttpInterface httpInterface, String playerScript, String dashUrl) throws Exception;
}

File: main/src/main/java/lavaplayer/source/youtube/YoutubeSignatureResolver.kt
package lavaplayer.source.youtube

import lavaplayer.tools.io.HttpInterface
import java.io.IOException
import java.net.URI


interface YoutubeSignatureResolver {
    @Throws(Exception::class)
    fun resolveFormatUrl(httpInterface: HttpInterface, playerScript: String?, format: YoutubeTrackFormat): URI

    @Throws(Exception::class)
    fun resolveDashUrl(httpInterface: HttpInterface, playerScript: String?, dashUrl: String): String

    @Throws(IOException::class)
    fun getCipherKeyAndTimestampFromScript(httpInterface: HttpInterface, playerScript: String): YoutubeSignatureCipher?
}
